### DISCUSSION TITLE
Dynamically Assign commitMessagePrefix using postUpgradeTasks

### DISCUSSION THREAD

**Discussion by jacob-jameson** [renovatebot/renovate#22422] [Request Help]
### How are you running Renovate?

Self Hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

renovate/renovate:latest

### If you're self-hosting Renovate, select which platform you are using.

GitHub Enterprise Server

### If you're self-hosting Renovate, tell us what version of the platform you run.

GitHub Enterprise Server 3.5.11

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I have created a self-hosted Renovate bot for my team at work, and it is working fine. Part of our company's path to production standards is that each commit to a deployable branch has to reference a Jira Issue in the commit message. Currently, we comply with this standard by having a long term open Jira Issue and we hard code the issue code into the `commitMessagePrefix` config field.

We would like to create a new Jira issue for each pull request that Renovate creates, and dynamically set the `commitMessagePrefix` to equal the new Jira issue code.

### What you tried so far.

I have created a bash script that creates a new Jira issue using their REST api, the new issue's code is saved into an environment variable. 

Let's say my script is called `Jira.sh` and the environment variable is called `JiraCode`. 

In the server side configuration I set 
```
allowedPostUpgradeCommands: ["Jira.sh"],
customEnvVariables:{"JiraCode":""}
```
In the repo configuration I set 
```
"postUpgradeTasks":{"commands": ["Jira.sh"], "fileFilters": [], "executionMode": "branch"},
"commitMessagePrefix":"{env.JiraCode}"
```

At the moment this is resulting in the empty string being used as the commit message prefix.

I know that commands listed in `postUpgradeTasks` are executed before Renovate commits the changes. So, in theory, it should be possible to use them to dynamically assign Jira issue codes to the  `commitMessagePrefix` config field. Is there a different approach that would give me the behavior I want?

### Relevant debug logs

I am confident that my script has executed successfully, as it passes all of my run time tests. The renovate debug logs also indicate that my script ran as expected, as seen in the log snippet below.

There are a lot of log lines that I cannot post here, so that I comply with my company's security and copyright standards. But, I think for this discussion, the logs snippet I have posted should be sufficient.

<details><summary>Logs</summary>

```
DEBUG: Executed post-upgrade task (REDACTED FOR PRIVACY)
       "dep": "REDACTED FOR PRIVACY",
       "cmd": "Jira.sh",
       "stderr": "",
       "stdout": "TASK: Post Upgrade Task Was Succesfully Executed!!\n"
```

</details>

**Comment by rarkins** [comment]
I don't think this is possible without a lot of rework

**Reply by jacob-jameson** [reply]
Does Renovate use some kind of git commit message template file under the hood? Because, If I found a way of editing that template file I could add a trailer with the Jira code in it. The Jira code doesn't have to be a prefix of the commit message, it just needs to be referenced somewhere in the message.

**Reply by rarkins** [reply]
It's all application logic and a child process task cannot change application memory or logic

**Reply by jacob-jameson** [reply]
That makes sense.

Personally, I think allowing postUpgradeTasks to modify commit messages and stuff like that would be a good feature to have. But, it is somewhat of a niche use case of postUpgradeTasks.

**Reply by viceice** [reply]
A probably easier way to solve it is to create a seperate CI job, which creates the jira issue and amends the renovate commit when renovate pushed a new commit. Use `ignoreAuthors` to let renovate ignore your commit author.

**Reply by jacob-jameson** [reply]
Yeah, that's what I'm going to do. I does mean that we need to rebase commits, which is less than ideal, but it gets the necessary behavior
